### ISSUE TITLE
Code fulfills the challenge requirements but fails to pass

### ISSUE THREAD

**Issue by peter325** [FreeCodeCamp/FreeCodeCamp#9814]
Challenge [Return Largest Numbers in Arrays](https://www.freecodecamp.com/challenges/return-largest-numbers-in-arrays#?solution=var%20res%20%3D%5B%5D%3B%0A%0Afunction%20largestOfFour(arr)%20%7B%0Afunction%20highlow(a%2Cb)%20%7B%0A%20%20%20%20return%20b%20-%20a%3B%0A%7D%0A%20%20for(i%3D0%3B%20i%3C%20arr.length%3B%20i%2B%2B)%7B%0A%20%20arr%5Bi%5D.sort(highlow)%3B%0A%20%20%7D%0A%20%20for(j%3D0%3B%20j%20%3C%20arr.length%3B%20j%2B%2B)%7B%0A%20%20res.push(arr%5Bj%5D%5B0%5D)%3B%0A%20%20%7D%0A%20%20return%20res%3B%0A%7D%0A%0AlargestOfFour(%5B%5B13%2C%2027%2C%2018%2C%2026%5D%2C%20%5B4%2C%205%2C%201%2C%203%5D%2C%20%5B32%2C%2035%2C%2037%2C%2039%5D%2C%20%5B1000%2C%201001%2C%20857%2C%201%5D%5D)%3B%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36</code>.

Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```javascript
var res =[];

function largestOfFour(arr) {
function highlow(a,b) {
    return b - a;
}
  for(i=0; i< arr.length; i++){
  arr[i].sort(highlow);
  }
  for(j=0; j < arr.length; j++){
  res.push(arr[j][0]);
  }
  return res;
}

largestOfFour([[13, 27, 18, 26], [4, 5, 1, 3], [32, 35, 37, 39], [1000, 1001, 857, 1]]);

```
The code meets all the criteria and should pass but the green ticks do no appear.

> <img src="http://oi67.tinypic.com/6qh8nq.jpg.">

**Comment by BKinahan** [comment]
Because of the way the tests are run, with multiple calls to your function in a row, any global variables you use will not be cleared between tests and so correct code will only provide the correct output for the very first test.

This will also result in the code not working correctly for multiple function calls in any environment, so the best solution is to move the global variable declaration (`var res =[];`) to the inside of your function so that it is initialized as an empty array each time the function is called, and to do this for each function you write unless such a side-effect is desired (see Counting Cards).

In short, this is not an issue with the challenge, but an easily-fixed issue with your code. 

Happy coding!

**Comment by raisedadead** [comment]
@peter325 thanks for reporting but we are currently discussing the use of global vars, please follow along on #9199

You can however refactor your code to pass the challenge without using a global variable.

**Comment by peter325** [comment]
Thanks for your clear and concise explanation that give me a more in-depth understand much appropriated!
